local-tests/setup/shiva-client.ts

```diff
@@ -102,100 +110,82 @@ export class TestnetClient {
+   * Returns the config for a given testnet
    */
-  public getTestnetConfig() {
-    return fetch(
+  public async getTestnetConfig() {
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL +
         '/test/get/info/testnet/' +
         this._id
-    )
-      .then((res: Response) => {
-        return res.json();
-      })
-      .then((info: TestNetResponse<TestNetInfo>) => {
-        this._info = info.body;
-        this._currentState = info.lastStateObserved as TestNetState;
-        console.log('setting testnet info: ', this._info);
-      });
+    );
+
+    let testnetConfigRes = _processTestnetResponse<boolean>(res);
+
+    return testnetConfigRes;
   }
 
   /**
    * Will wait for the NEXT epoch and return a resposne when the epoch has fully transitioned.
    * The return time is directly proportional to the epoch transition time config and where the network is with the current epoch.
    */
-  public transitionEpochAndWait() {
-    return fetch(
+  public async transitionEpochAndWait() {
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL +
         '/test/action/transition/epoch/wait/' +
         this._id
-    )
-      .then((res: Response) => {
-        if (res.status === 200) {
-          return res.json();
-        }
-      })
-      .then((body: any) => {
-        console.log('Stopped random peer: ', body);
-      });
+    );
+
+    let transitionEpochAndWaitRes = _processTestnetResponse<boolean>(res);
+
+    return transitionEpochAndWaitRes;
   }
 
   /**
    * Stops a random peer and waits for the next epoc to transiton.
    * The return time is directly proportional to the epoch transition time config and where the network is with the current epoch.
    */
-  public stopRandomNetworkPeerAndWaitForNextEpoch() {
-    return fetch(
+  public async stopRandomNetworkPeerAndWaitForNextEpoch() {
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL +
         '/test/action/stop/random/wait/' +
         this._id
-    )
-      .then((res: Response) => {
-        if (res.status === 200) {
-          return res.json();
-        } else {
-          throw res;
-        }
-      })
-      .then((body: TestNetResponse<boolean>) => {
-        console.log('validator kick response: ', body);
-        return body;
-      });
+    );
+
+    let randomPeerStopRes = _processTestnetResponse<boolean>(res);
+
+    return randomPeerStopRes;
   }
 
   /*
     Stops the testnet
   */
-  public stopTestnet() {
+  public async stopTestnet() {
     console.log('stopping testnet with id:', this._id);
-    return fetch(
+    const res = await fetch(
       this._processEnvs.TESTNET_MANAGER_URL + '/test/delete/testnet/' + this._id
-    )
-      .then((res: Response) => {
-        return res.json();
-      })
-      .then((body: TestNetResponse<void>) => {
-        console.log('shutdown respone: ', body);
-      });
+    );
+
+    let testnetResp = _processTestnetResponse<boolean>(res);
```

Comment:
hmm -- stashing the promise returned by `_processTestnetResponse()` in a local variable here is a little unexpected; if you inspected it using a debugger on L188, it would not yet be resolved/rejected, and we don't `await` it here.  Maybe just `return _processTestnetResponse<bolean>(res);` here? Or `await` it if you want the local variable in scope for some purpose